### ISSUE TITLE
The signature of MAIN_HELPER has changed

### ISSUE THREAD

**Issue by zostay** [zostay/p6-Getopt-ForClass#2]
This module overrides `MAIN_HELPER`, which is the not-really-documented way in which the arguments and handling of `MAIN` is performed. This appears to be part of the 6.d changes to help support the new way `$*ARGFILES` are being handled.

I am thinking the best way to support this is to have this module support 6.c until the 6.d release comes out and then make a new 1.0 release that supports 6.d.

**Comment by zostay** [comment]
No, now I'm disagreeing with myself... there are a couple other possible solutions that will allow us to remain compatible with both, I think:

* Implement `MAIN_HELPER` as a `multi` with one taking one argument and the other taking two.
* Implement an export helper that detects whether this is 6.c or 6.d and export the correct `MAIN_HELPER` for that version of Perl.

**Comment by zostay** [comment]
The current master branch contains an experiment using `multi`. This change works on with the new `MAIN_HELPER` signature, but not with the old one, where it causes an ambiguous method exception.=

**Comment by Leont** [comment]
The handling of this has changed, again.
